wearApp/src/main/java/dev/johnoreilly/confetti/wear/bookmarks/BookmarksComponent.kt

```diff
@@ -0,0 +1,74 @@
+package dev.johnoreilly.confetti.wear.bookmarks
+
+import com.apollographql.apollo3.cache.normalized.FetchPolicy
+import com.arkivanov.decompose.ComponentContext
+import dev.johnoreilly.confetti.ConfettiRepository
+import dev.johnoreilly.confetti.GetBookmarkedSessionsQuery
+import dev.johnoreilly.confetti.auth.User
+import dev.johnoreilly.confetti.decompose.coroutineScope
+import dev.johnoreilly.confetti.toTimeZone
+import dev.johnoreilly.confetti.utils.ClientQuery.toUiState
+import dev.johnoreilly.confetti.utils.QueryResult
+import dev.johnoreilly.confetti.utils.nowAtTimeZone
+import dev.johnoreilly.confetti.wear.complication.ComplicationUpdater
+import dev.johnoreilly.confetti.wear.tile.TileUpdater
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.datetime.LocalDateTime
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+interface BookmarksComponent {
```

Comment:
Why do we need separate components for Wear? Can we reuse the existing ones?

Reply:
I did except when the screen structure meant they were quite different.

I will review this one again and reuse if possible.

For some of the common ones I do need to follow up and make them less network aggressive in a per device way. So I want to share.

Reply:
Yeah, maybe generalise some of the existing components, so that they could be reused for Wear. It should be fine to have some parts unused on Wear. Some things could be abstracted with an interface. But if the logic is really different, then it might be better to actually have them separate. So up to you.

I think you are lucky that you only replaced top components. Replacing an intermediate component (in the middle of the hierarchy) would be a bit tricker.

Reply:
I think a follow up. The current one has slight differences in screen structure. Would like to get this in first before refactoring mobile further.

Reply:
Sounds good.